README.md

```diff
@@ -45,51 +52,85 @@ Usage: dandi [OPTIONS] COMMAND [ARGS]...
+  instances         List known Dandi Archive instances that the CLI can...
+  ls                List .nwb files and dandisets metadata.
+  move              Move or rename assets in a local Dandiset and/or on...
+  organize          (Re)organize files according to the metadata.
+  service-scripts   Various utility operations
+  shell-completion  Emit shell script for enabling command completion.
+  upload            Upload Dandiset files to DANDI Archive.
+  validate          Validate files for NWB and DANDI compliance.
+  validate-bids     Validate BIDS paths.
 ```
 
-Each of the commands has a set of options to alter their behavior.  Please run
-`dandi COMMAND --help` to get more information, e.g.
+Each of the commands has a set of options to alter its behavior.  Run
+`dandi COMMAND --help` to get more information:
 
 ```
 $> dandi ls --help
-Usage: dandi ls [OPTIONS] [PATHS]...
-
-  List .nwb files metadata
+Usage: dandi ls [OPTIONS] PATH|URL
+
+  List .nwb files and dandisets metadata.
+
+  The arguments may be either resource identifiers or paths to local
+  files/directories.
+
+  Accepted resource identifier patterns:
+   - DANDI:<dandiset id>[/<version>]
+   - https://dandiarchive.org/...
+   - https://identifiers.org/DANDI:<dandiset id>[/<version id>] (<version id> cannot be 'draft')
+   - https://<server>[/api]/[#/]dandiset/<dandiset id>[/<version>][/files[?location=<path>]]
+   - https://*dandiarchive-org.netflify.app/...
+   - https://<server>[/api]/dandisets/<dandiset id>[/versions[/<version>]]
+   - https://<server>[/api]/assets/<asset id>[/download]
+   - https://<server>[/api]/dandisets/<dandiset id>/versions/<version>/assets/<asset id>[/download]
+   - https://<server>[/api]/dandisets/<dandiset id>/versions/<version>/assets/?path=<path>
+   - dandi://<instance name>/<dandiset id>[@<version>][/<path>]
+   - https://<server>/...
 
 Options:
   -F, --fields TEXT               Comma-separated list of fields to display.
                                   An empty value to trigger a list of
                                   available fields to be printed out
-  -f, --format [auto|pyout|json|json_pp|yaml]
+  -f, --format [auto|pyout|json|json_pp|json_lines|yaml]
                                   Choose the format/frontend for output. If
                                   'auto', 'pyout' will be used in case of
                                   multiple files, and 'yaml' for a single
                                   file.
+  -r, --recursive                 Recurse into content of
+                                  dandisets/directories. Only .nwb files will
+                                  be considered.
+  -J, --jobs INTEGER              Number of parallel download jobs.  [default:
+                                  6]
+  --metadata [api|all|assets]
+  --schema VERSION                Convert metadata to new schema version
   --help                          Show this message and exit.
 ```
```

Comment:
It may be out of scope for this PR, but this section seems like it can be simplified to letting the user know that they can run `dandi --help` and `dandi <subcommand> --help` to get manpages. Listing sample output means we'll need to keep updating this section every time the CLI changes its interface.

Reply:
That had crossed my mind too, wasn't sure how often this changed. Added to my Phase 2 docs backlog for 2023.